test/gtest/uct/test_uct_perf.cc

```diff
@@ -15,67 +15,72 @@ extern "C" {
-                                        "SandyBridge far socket";
-                    check_perf = false;
-                    break;
-                }
+void test_uct_perf::test_execute(unsigned flags = 0,
+                                 ucs_memory_type_t send_mem_type =
+                                 UCS_MEMORY_TYPE_HOST,
+                                 ucs_memory_type_t recv_mem_type =
+                                 UCS_MEMORY_TYPE_HOST) {
+    if (has_transport("ugni_udt")) {
+        UCS_TEST_SKIP;
+    }
+
+    /* For SandyBridge CPUs, don't check performance of far-socket devices */
+    std::vector<int> cpus = get_affinity();
+    bool check_perf       = true;
+    size_t max_iter       = std::numeric_limits<size_t>::max();
+
+    if (ucs_arch_get_cpu_model() == UCS_CPU_MODEL_INTEL_SANDYBRIDGE) {
+        for (std::vector<int>::iterator iter = cpus.begin();
+             iter != cpus.end(); ++iter) {
+            if (!ucs_cpu_is_set(*iter, &GetParam()->local_cpus)) {
+                UCS_TEST_MESSAGE << "Not enforcing performance on "
+                                    "SandyBridge far socket";
+                check_perf = false;
+                break;
             }
         }
+    }
 
-        if (has_transport("tcp")) {
-            check_perf = false; /* TODO calibrate expected performance based on transport */
-            max_iter   = 1000lu;
-        }
+    if (has_transport("tcp")) {
+        check_perf = false; /* TODO calibrate expected performance based on transport */
+        max_iter   = 1000lu;
+    }
 
-        /* Run all tests */
-        for (const test_spec *test_iter = tests; test_iter->title != NULL;
-             ++test_iter) {
-            test_spec test = *test_iter;
-
-            test.send_mem_type = UCS_MEMORY_TYPE_HOST;
-            test.recv_mem_type = UCS_MEMORY_TYPE_HOST;
-
-            if (has_transport("cuda_copy") || has_transport("gdr_copy")) {
-                test.max *= UCT_CUDA_PERF_TEST_MULTIPLIER;
-                test.min /= UCT_CUDA_PERF_TEST_MULTIPLIER;
-
-                test.recv_mem_type = UCS_MEMORY_TYPE_CUDA;
-            } else if (ucs_arch_get_cpu_model() == UCS_CPU_MODEL_ARM_AARCH64) {
-                test.max *= UCT_ARM_PERF_TEST_MULTIPLIER;
-                test.min /= UCT_ARM_PERF_TEST_MULTIPLIER;
-            } else {
-                test.max *= UCT_PERF_TEST_MULTIPLIER;
-                test.min /= UCT_PERF_TEST_MULTIPLIER;
-            }
+    /* Run all tests */
+    for (const test_spec *test_iter = tests; test_iter->title != NULL;
+         ++test_iter) {
+        test_spec test = *test_iter;
+
+        test.send_mem_type = send_mem_type;
+        test.recv_mem_type = recv_mem_type;
+
+        if (has_transport("cuda_copy")) {
+            test.max *= UCT_CUDA_PERF_TEST_MULTIPLIER;
+            test.min /= UCT_CUDA_PERF_TEST_MULTIPLIER;
+        } else if (ucs_arch_get_cpu_model() == UCS_CPU_MODEL_ARM_AARCH64) {
```

Comment:
I would remove this 'else'. so on Arm platform, the min/max would be multiplied both by UCT_CUDA_PERF_TEST_MULTIPLIER and UCT_ARM_PERF_TEST_MULTIPLIER. I think it doesn't make sense they are mutually exclusive.